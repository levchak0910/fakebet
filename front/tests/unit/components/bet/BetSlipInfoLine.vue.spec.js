import {createWrapper, disableFile, cutFromOptions, mapProperties} from "../../__utils__";
import Component from "@/components/bet/betSlip/Info/Line";

disableFile();

describe("bet/BetSlipInfoLine.vue", () => {

    let wrapper, closeButton;
    let mountWrapper = (options = {}) => {
        let {props} = cutFromOptions(options, ["props"]);
        wrapper = createWrapper(Component, {
            props: {
                bookie: {coefficient: 2},
                ...props
            },
            ...options,
        });
        closeButton = wrapper.find(".close");
    };

    it("Testing snapshot", () => {
         mountWrapper({
            props: {
                ...mapProperties("dateTmpl", "league", "home", "guest", "type"),
                bookie: {name: "{{bookie.name}}", coefficient: 2},
            },
            computed: {
                show: mapProperties("waiting", "results", true),
                ...mapProperties("coefficientTmpl", "betClass", "scoreTmpl"),
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    describe("Testing triggering methods", () => {

        it("deleteBet", () => {
            mountWrapper({
                methods: ["deleteBet"],
                computed: {
                    show: {waiting: true},
                },
            });

            closeButton.trigger("click");

            expect(wrapper.vm.deleteBet).toBeCalled();
        });

    });

    describe("Testing computed properties", () => {

        describe("show", () => {

            it("show = waiting", () => {
                mountWrapper({
                    props: {
                        lineMode: "waiting",
                    },
                });

                expect(wrapper.vm.show).toEqual({waiting: true, results: false});
            });

            it("show = results", () => {
                mountWrapper({
                    props: {
                        lineMode: "result",
                    },
                });

                expect(wrapper.vm.show).toEqual({waiting: false, results: true});
            });

        });

        it("coefficientTmpl", () => {
            mountWrapper({
                props: {
                    bookie: {coefficient: 26.577},
                },
            });

            expect(wrapper.vm.coefficientTmpl).toBe("26.58");
        });

        describe("scoreTmpl", () => {

            it("scoreTmpl = result", () => {
                mountWrapper({
                    props: {
                        score: "3 : 1",
                        matchResult: "1",
                    },
                });

                expect(wrapper.vm.scoreTmpl).toBe("3 : 1");
            });

            it("scoreTmpl = default template", () => {
                mountWrapper({
                    props: {
                        matchResult: null,
                    },
                });

                expect(wrapper.vm.scoreTmpl).toBe("- : -");
            });

        });

        describe("betClass", () => {

            it("betClass = empty", () => {
                mountWrapper({
                    props: {
                        matchResult: null,
                    },
                });

                expect(wrapper.vm.betClass).toBe("");
            });

            it("betClass when win", () => {
                mountWrapper({
                    props: {
                        matchResult: "1",
                        type: "1",
                    },
                });

                expect(wrapper.vm.betClass).toBe("bet-match-won");
            });

            it("betClass when lose", () => {
                mountWrapper({
                    props: {
                        matchResult: "2",
                        type: "1",
                    },
                });

                expect(wrapper.vm.betClass).toBe("bet-match-lose");
            });

        });

    });

    describe("Testing methods", () => {

        it("deleteBet", () => {
            mountWrapper({
                props: {
                    betIndex: 1,
                    betSlipIndex: 2,
                },
                methods: ["_deleteBet"],
            });

            wrapper.vm.deleteBet();

            expect(wrapper.vm._deleteBet).toBeCalledWith({indexOfBet: 1, indexOfBetSlip: 2});
        });

    });

});
