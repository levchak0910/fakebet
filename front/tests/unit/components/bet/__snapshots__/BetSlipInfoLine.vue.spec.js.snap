// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bet/BetSlipInfoLine.vue Testing snapshot 1`] = `
<div
  class="row"
>
  <span
    class="close"
  >
    ✖
  </span>
  <span
    class="date"
  >
    {{dateTmpl}}
  </span>
  <span
    class="league"
  >
    {{league}}
  </span>
  <span
    class="teams"
  >
    {{home}} — {{guest}}
  </span>
  <span
    class="type"
  >
    {{type}}
  </span>
  <span
    class="coefficient"
    title="bookie 1"
  >
    {{coefficientTmpl}}
  </span>
  <span
    class="score {{betClass}}"
  >
    {{scoreTmpl}}
  </span>
</div>
`;
